src/CheckBox/index.jsx

```diff
@@ -36,8 +36,8 @@ class CheckBox extends React.Component {
         checked: true,
       });
     }
-    if (this.onChange) {
-      this.onChange();
+    if (this.onChangeState) {
+      this.onChangeState();
```

Comment:
I'm also curious what would happen if you try just commenting out this `handleClick` method entirely. I built `asInput` with the intention that it would handle all change events for its descendant components, so the descendants wouldn't have to mess around with their own logic at all. Does `asInput`'s `handleChange` fire when this checkbox's value changes?